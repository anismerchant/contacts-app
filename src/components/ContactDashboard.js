import React from 'react';

const ContactDashboard = () => (
    <div>
        Contact Dashboard
    </div>
);

export default ContactDashboard;