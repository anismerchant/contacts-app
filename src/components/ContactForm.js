import React, { Component } from 'react';

class ContactForm extends Component {
    render() {
        return (
            <form>
                <input 
                    type="text"
                    placeholder="First Name"
                />
                <input 
                    type="text"
                    placeholder="Last Name"
                />
                <input 
                    type="email"
                    placeholder="Email" 
                    id="email"
                    pattern=".+@globex.com" 
                    size="30" 
                />
                <input 
                    type="tel"
                    placeholder="Cell" 
                    id="cell" 
                    name="cell"
                    pattern="[0-9]{3}-[0-9]{3}-[0-9]{4}"
                />
            </form>
        );
    }
}

export default ContactForm;