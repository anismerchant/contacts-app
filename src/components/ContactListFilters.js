import React from 'react';
import { connect } from 'react-redux';
import { setNameFilter, sortByFirstName, sortByLastName } from '../actions';

const ContactListFilters = (props) => (
    <div className="contactlist--filters">
        <input 
            type="text" 
            placeholder="Search"
            value={props.filters.name} 
            onChange={(e) => props.dispatch(setNameFilter(e.target.value))}
        />
        <select
            value={props.filters.sortBy}
            onChange = {(e) => 
            {
                if (e.target.value === "firstname") {
                    props.dispatch(sortByFirstName())
                } else if (e.target.value === "lastname") {
                    props.dispatch(sortByLastName())
                } 
            }}
        >   
            <option className="contactlist--sortby" value="" disabled>Sort by</option>
            <option className="contactlist--firstname" value="firstname">First Name</option>
            <option className="contactlist--lastname" value="lastname">Last Name</option>
        </select>
    </div>

);

const mapStateToProps = (state) => {
    return {
        filters: state.filters
    };
};

export default connect(mapStateToProps)(ContactListFilters);
