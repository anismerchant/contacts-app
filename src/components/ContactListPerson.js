import React from 'react';
import { removeContact } from '../actions/contacts';
import { connect } from 'react-redux';

// de-structured props
const ContactListPerson = ({dispatch, id, firstname, lastname, email, cell}) => (
    <div>
        <p>{`${firstname} ${lastname} ${email} ${cell}`}</p>
        <button onClick={ () => {dispatch(removeContact({id}))
        }}>Remove</button>
    </div>
);

export default connect()(ContactListPerson);