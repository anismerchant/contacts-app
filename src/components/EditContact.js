import React from 'react';

const EditContact = () => (
    <div>
        Edit Contact Page
    </div>
);

export default EditContact;