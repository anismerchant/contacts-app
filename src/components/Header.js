import React from 'react';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const Header = ({startLogout}) => (
    <header>
        <h1> Contacts App</h1>
        <button onClick={startLogout}>Logout</button>
    </header>
);

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);