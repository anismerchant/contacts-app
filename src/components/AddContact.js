import React from 'react';
import { connect } from 'react-redux';
import ContactForm from './ContactForm';
import { beginAddContact }  from '../actions/contacts'

export const AddContact = (props) => {
    return (
        <div>
            <h3>Add Contact</h3>
            <ContactForm 
                onSubmit={ (contact) => {
                    props.dispatch(beginAddContact(contact));
                    props.history.push('/dashboard');
                }}
            />
        </div>
    );
}

export default connect()(AddContact);