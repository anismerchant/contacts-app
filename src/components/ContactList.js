import React from 'react';
import { connect } from 'react-redux';
import ContactListPerson from './ContactListPerson';
import findContacts from '../selectors/findContacts';

// de-structured props
const ContactList = ({contacts}) => (
    <div>
        {contacts.length}
        {contacts.map((contact) => <ContactListPerson key={contact.id} {...contact} />)}
    </div>
);

const mapStateToProps = (state) => {
    return {
        contacts: findContacts(state.contacts, state.filters)
    }
}

export default connect(mapStateToProps)(ContactList);