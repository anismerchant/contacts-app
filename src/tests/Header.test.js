import React from 'react';
import { shallow } from 'enzyme';
import { Header } from '../components/Header';

test('It should correctly render Header.', () => {
    const wrapper = shallow(<Header startLogout={() => { }} />);
    expect(wrapper).toMatchSnapshot();
});

test('It should call startLogout on button click.', () => {
    const startLogout = jest.fn();
    const wrapper = shallow(<Header startLogout={startLogout} />)
    wrapper.find('button').simulate('click');
    expect(startLogout).toHaveBeenCalled();
});