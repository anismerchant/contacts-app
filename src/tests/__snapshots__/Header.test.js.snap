// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should correctly render Header. 1`] = `
<header>
  <h1>
     Contacts App
  </h1>
  <button
    onClick={[Function]}
  >
    Logout
  </button>
</header>
`;
